examples/with-okta-multi-tenant-pkce-flow/api-server/okta.go

```diff
@@ -0,0 +1,144 @@
+package main
+
+import (
+	"crypto/rand"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"sync"
+
+	"github.com/supertokens/supertokens-golang/recipe/thirdparty/tpmodels"
+	"github.com/supertokens/supertokens-golang/supertokens"
+
+	verifier "github.com/nirasan/go-oauth-pkce-code-verifier"
+)
+
+type OktaConfig struct {
+	ClientID     string
+	ClientSecret string
+	OktaDomain   string
+}
+
+var oktaVerifiers = sync.Map{}
+
+func newState() string {
+	const chars = "0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ"
+	bytes := make([]byte, 64)
+	if _, err := rand.Read(bytes); err != nil {
+		return ""
+	}
+
+	for i, b := range bytes {
+		bytes[i] = chars[b%byte(len(chars))]
+	}
+	return string(bytes)
+}
+
+var oktaProvider = tpmodels.TypeProvider{
+	ID:        "okta",
+	IsDefault: true,
+	Get: func(redirectURI, authCodeFromRequest *string, userContext supertokens.UserContext) tpmodels.TypeProviderGetResponse {
+		config, _ := (*userContext)["oktaConfig"].(OktaConfig)
+		// tenantId, _ := (*userContext)["tenantId"].(string)
+		oktaRedirectURI, _ := (*userContext)["redirectURI"].(string)
+
+		if authCodeFromRequest == nil {
+			empty := ""
+			authCodeFromRequest = &empty
+		}
+
+		state, _ := (*userContext)["state"].(string)
```

Comment:
So explain what this state is and why it's needed

Reply:
done